src/components/Tooltip/index.js

```diff
@@ -0,0 +1,202 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './style.css';
+
+import Icon from '../Icon';
+
+export const directions = ['top', 'left', 'right', 'bottom'];
+
+const Tooltip = ({
+  children,
+  closeable,
+  closeOnEscape,
+  content: tooltipContent,
+  direction,
+  isOpen,
+  showOnHover,
+  tooltipProps: tooltipOptsProp,
+  tooltipWidth,
+  ...rest
+}) => {
+  const [tooltipId] = useState(
+    `tooltip-${Date.now() * Math.floor(100 * Math.random())}`
```

Comment:
Curiosity: How'd you land on this ID generator? Was random not random enough in practice?

Reply:
Well, I wanted to avoid just using numbers. And since everything could get rendered almost at the time time, I wanted to avoid any accessibility clashes. I figured multiplying the date by a random number would be definitively random.

Reply:
Well, I wanted to avoid just using numbers. And since everything could get rendered almost at the time time, I wanted to avoid any accessibility clashes. I figured multiplying the date by a random number would be definitively random.